### PR TITLE
Add '~(a - b) -> ((b - a) - 1)' simplification rule

### PR DESCRIPTION
**Pull request by ptomin**

### CODE REVIEW

**Review by uxmal — COMMENTED** [review]
Minor nit, otherwise gtg.

File: src/Decompiler/Evaluation/CompSub_Rule.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
+#region License
+/* 
+ * Copyright (C) 1999-2022 Pavel Tomin.
```
[the remark this answers is not in the document]
Comment (ptomin) [reply src/Decompiler/Evaluation/CompSub_Rule.cs:3]: Oh, I have started this implementation last year! Will be fixed soon.

[the remark this answers is not in the document]
Comment (ptomin) [reply src/Decompiler/Evaluation/CompSub_Rule.cs]: Done in 849cb34

**Comment by uxmal** [comment]
Looks good!

**Pull request merged**